src/librustdoc/html/render/print_item.rs

```diff
@@ -1418,3 +1424,27 @@ fn document_non_exhaustive(w: &mut Buffer, item: &clean::Item) {
         w.write_str("</div>");
     }
 }
+
+fn document_ty_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
+    let param_env = cx.tcx().param_env(ty_def_id);
+    let ty = cx.tcx().type_of(ty_def_id);
+    match cx.tcx().layout_of(param_env.and(ty)) {
+        Ok(ty_layout) => {
+            writeln!(w, r#"<h2 class="small-section-header">Layout</h2>"#);
+            writeln!(w, "<div>");
+            if ty_layout.layout.abi.is_unsized() {
+                writeln!(w, "<strong>Sized:</strong> (unsized)");
+            } else {
+                writeln!(
+                    w,
+                    "<strong>Size:</strong> {size} byte{pl}",
+                    size = ty_layout.layout.size.bytes(),
+                    pl = if ty_layout.layout.size.bytes() == 1 { "" } else { "s" },
+                );
+            }
+            writeln!(w, "</div>");
+        }
+        // FIXME: should we crash instead? or report an error?
+        Err(_layout_err) => {}
```

Comment:
Can you do this just temporarily so I can get an idea of when this happens? I don't know why `layout_of` would return an error.

Reply:
Based on looking at the docs, the two possible errors that can be returned are ``the type `{}` has an unknown layout`` and ``values of the type `{}` are too big for the current architecture``. So I *think* those will only be encountered if there's a problem in the user's code. Running a build now to see what happens if we panic.

Reply:
> So I _think_ those will only be encountered if there's a problem in the user's code.

Nope, it turns out there are definitely cases where you get a layout error but the code is fine. Specifically, it seems like any generic type will give you a `LayoutError::Unknown` because it doesn't know the size of the generic parameter. For example, [`Result<T, E>` gives an error](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e28ff4560b78baa2427ccb922d716f6b). I think this explains why many of the types in the standard library didn't have layout info—it's because they're generic.

I'm not sure if there's anything we can do about this to be able to show the layout of generic types. Maybe we'll still add the layout section but just put "layout is unknown due to generic parameters" or something like that. What do you think?

Reply:
I think we should just omit the type size in that case, that seems fine.